src/Passes/Source/Passes/TransformationRule/LibTransformationRule.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Passes/TransformationRule/TransformationRule.hpp"
+#include "Rules/Factory.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <fstream>
+#include <iostream>
+
+namespace
+{
+llvm::PassPluginLibraryInfo getTransformationRulePluginInfo()
+{
+    using namespace microsoft::quantum;
+    using namespace llvm;
+
+    return {LLVM_PLUGIN_API_VERSION, "TransformationRule", LLVM_VERSION_STRING, [](PassBuilder& pb) {
+                // Registering the pass
+                pb.registerPipelineParsingCallback(
+                    [](StringRef name, FunctionPassManager& fpm, ArrayRef<PassBuilder::PipelineElement> /*unused*/) {
+                        // Base profile
```

Comment:
Could you elaborate a bit more regarding what is at this time configurable, and what is still hardcoded? Is any kind of profile definition as yml file supported at this time, or is this PR containing only the hardcoded handling of the base profile specifically?

Reply:
Everything that is done with our own tool `qat` is configurable and everything with `opt` is hard coded. As such, I suspect we will deprecate the usage of `opt` over time but it is quite instructive to have both to test certain aspects of a pass and to run passes individually (even if they are not configurable). Whether we want to make a pass for every profile is questionable though as it provides a lot of extra maintenance with little extra benefit.